trunner/test/harnesses/test_micropython.py

```diff
@@ -0,0 +1,503 @@
+        matching_texts = [
+            '''start\r\nstop''',
+
+            '''start\r\nline1\r\nstop''',
+
+            '''start\r\nline1\r\nline2\r\nstop''',
+
+            '''start\r\nline1\r\r\nline2\r\r\r\nstop'''
+        ]
+
+        create_and_check_regex(input, matching_texts)
+
+
+class TestStandardHarness:
+
+    # Program for communicating with harness
+    def fake_program(socket, args):
+        test_path = args[0]
+        program_output = args[1]
+        exp_output = args[2]
+
+        send_through_socket(socket, program_output)
+        # b = program_output.encode("utf-8")
+        # socket.send(b)
+
+        if exp_output is not None:
+            recv_exp_through_socket(socket, f"cat {test_path}.exp\n")
+            send_through_socket(socket, exp_output)
+
+    def test_standard_pass(self):
+        test_path = "/usr/test/micropython/basics/test.py"
+        program_output = f"Running test: {test_path}\r\nLorem ipsum{PROMPT}"
+        exp_output = f"cat {test_path}.exp\r\nLorem ipsum{PROMPT}"
+
+        result = get_harness_output(MicropythonStandardHarness().harness,
+                                    TestStandardHarness.fake_program,
+                                    (test_path, program_output, exp_output))
+
+        check_result(result, TestResult(test_path, TestResult.PASS))
+
+    def test_long_pass(self):
+        test_path = "/usr/test/micropython/basics/test.py"
+
+        text = '''Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+        Nulla elementum, mauris pretium consectetur vestibulum, lectus nibh gravida velit,
+        ac interdum lectus ligula et quam.
+        Vestibulum ac nunc vel felis sodales bibendum vitae sit amet magna.
+        Curabitur elit mi, congue et dui vel, auctor pellentesque dolor.
+        Vestibulum lobortis placerat justo, iaculis placerat nibh dictum placerat.
+        Donec sapien leo, faucibus sit amet cursus eu, convallis eget erat.
+        Curabitur tincidunt nibh vel eros accumsan, eu vestibulum mi ullamcorper.'''
+
+        program_output = f'Running test: {test_path}\r\n{text}{PROMPT}'
+
+        exp_output = f'cat {test_path}.exp\r\n{text}{PROMPT}'
+
+        result = get_harness_output(MicropythonStandardHarness().harness,
+                                    TestStandardHarness.fake_program,
+                                    (test_path, program_output, exp_output))
+
+        check_result(result, TestResult(test_path, TestResult.PASS))
+
+    def test_program_error(self):
+        test_path = "/usr/test/micropython/basics/test.py"
+
+        program_name = "/bin/test_haness"
+        error_desc = "Some kind of error"
+        error = f"Error: {program_name} - {error_desc}"
+
+        program_output = f"Running test: {test_path}\r\n{error}"
+
+        with pytest.raises(AssertionError,
+                           match=f"There was an error {error_desc} during program {program_name} execution:\n{program_output}\n"):
```

Comment:
Just to avoid `line too long` warning.
```suggestion
        msg = f"There was an error {error_desc} during program {program_name} execution:\n{program_output}\n"
        with pytest.raises(AssertionError, match=msg):
```